wp-includes/class-wp-offline-page-excluder.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * WP_Offline_Page_Excluder class.
+ *
+ * @package PWA
+ */
+
+/**
+ * Class handles excluding the Offline Page's from the backend and frontend.
+ */
+class WP_Offline_Page_Excluder {
+
+	/**
+	 * Instance of the Offline Page Manager.
+	 *
+	 * @var WP_Offline_Page
+	 */
+	protected $manager;
+
+	/**
+	 * WP_Offline_Page_Filter constructor.
+	 *
+	 * @param WP_Offline_Page $manager Instance of the manager.
+	 */
+	public function __construct( $manager ) {
+		$this->manager = $manager;
+	}
+
+	/**
+	 * Initializes the instance.
+	 */
+	public function init() {
+		add_filter( 'wp_dropdown_pages', array( $this, 'exclude_from_page_dropdown' ), 10, 2 );
+		add_action( 'parse_query', array( $this, 'exclude_from_query' ) );
+	}
+
+	/**
+	 * Filters the HTML output of a list of pages as a drop down.
+	 *
+	 * @param string $html HTML output for drop down list of pages.
+	 * @param array  $args The parsed arguments array.
+	 * @return string Filtered content for wp_dropdown_pages.
+	 */
+	public function exclude_from_page_dropdown( $html, $args ) {
+		// Bail out if this is the offline page dropdown.
+		if ( WP_Offline_Page::OPTION_NAME === $args['name'] ) {
+			return $html;
+		}
+
+		return preg_replace( '/<option .*? value="' . $this->manager->get_offline_page_id() . '">.*?<\/option>/', '', $html );
+	}
+
+	/**
+	 * Exclude the offline page from the query when doing a search on the frontend or on the backend Menus page.
+	 *
+	 * @param WP_Query $query The WP_Query instance.
+	 */
+	public function exclude_from_query( WP_Query $query ) {
+		if ( $this->is_offline_page_query( $query ) ) {
+			$query->is_404      = true;
+			$query->is_page     = false;
+			$query->is_singular = false;
+			$query->set( 'page_id', 0 );
+		} elseif ( $this->is_okay_to_exclude( $query ) ) {
+			$query->set( 'post__not_in', array( $this->manager->get_offline_page_id() ) );
+		}
+	}
+
+	/**
+	 * Checks if the query is for the offline page.
+	 *
+	 * @param WP_Query $query The WP_Query instance.
+	 * @return bool Whether an offline page query.
+	 */
+	protected function is_offline_page_query( WP_Query $query ) {
+		if ( $query->is_admin ) {
+			return false;
+		}
+
+		if ( ! $query->is_page || ! $query->is_singular ) {
+			return false;
+		}
+
+		return ( $this->manager->get_offline_page_id() === $query->get( 'get_id' ) );
```

Comment:
Is `get_id` right here?

I think this method should largely just do:

```php
return $query->is_singular() && $this->manager->get_offline_page_id() === (int) $query->get_queried_object_id();
```

Reply:
@westonruter It is checking `is_singular` above in the guard clause:

```php
		if ( ! $query->is_page || ! $query->is_singular ) {
			return false;
		}
```

I'll switch that to `$query->is_singular()` instead of `$query->is_singular`.

Reply:
Note that `is_singular` implies `is_page`, so only one of the conditions is needed.

But I was more drawing attention to `$query->get( 'get_id' )`. I haven't seen `get_id` before. I don't believe that is right, and `get_queried_object_id()` is what you intend?